src/dune_rules/ordered_set_lang.mli

```diff
@@ -92,5 +92,33 @@ module Unexpanded : sig
 end
 with type expanded := t
 
+module Include_dirs : sig
+  type t
+
+  val eval : t -> Include_dir.t list
+
+  module Unexpanded : sig
+    type expanded = t
```

Comment:
Should be `type expanded := t` and you can remove the destructive substitution in the end.